lib/chef/provider/aws_iam_role.rb

```diff
@@ -0,0 +1,37 @@
+require 'chef/provisioning/aws_driver/aws_provider'
+
+class Chef::Provider::AwsIamRole < Chef::Provisioning::AWSDriver::AWSProvider
+  provides :aws_iam_role
+
+  def action_create
+    iam_role = super
+  end
+
+  protected
+
+  def create_aws_object
+    iam = new_resource.driver.iam_resource
+
+    converge_by "create IAM role #{new_resource.name}" do
+      iam.create_role({
+        path: new_resource.path,
+        role_name: new_resource.name,
+        assume_role_policy_document: new_resource.assume_role_policy_document
+      })
+    end
+  end
+
+  def update_aws_object(iam_role)
+  end
+
+  def destroy_aws_object(iam_role)
+    converge_by "delete #{iam_role.name}" do
+      iam_role.instance_profiles.each do |profile|
+        profile.remove_role(role_name: iam_role.name)
+      end
+      iam_role.policies.map(&:delete)
```

Comment:
Can the RolePolicies exist without the role?  Does AWS raise an error if you try to delete the Role while it still has RolePolicies?

Instead of doing this I think its better to do

```ruby
iam_role.policies.each do |policy|
  Cheffish.inline_resource(self, action) do
    aws_iam_role_policy policy do
      action :destroy
    end
  end
end
```